Automattic/jetpack
# Remove unnecessary WAF dependency in My Jetpack

This PR removes the unnecessary WAF package dependency in My Jetpack that was initially added in https://github.com/Automattic/jetpack/pull/38332.  Prior to this PR, the WAF package and all it's dependencies were being loaded into My Jetpack (for use in the Protect product card), and hence getting shipped to all our standalone plugins, which is not optimal.

See also Slack discussion: p1723736483790589-slack-C02LK1W8T4Z

Fixes: https://github.com/Automattic/jetpack-marketing/issues/932





## Proposed changes:

* Removed the WAF package from the My Jetpack package composer.json file.
* Conditionally call WAF_Runner::get_config() only if the WAF_Runner class exists (via the Protect plugin or Jetpack plugin).
    * If the WAF_Runner class does not exist, we can safely assume the Auto-firewall feature is off/disabled for the purposes of the Protect product card.

### Other information:


## Jetpack product discussion



The WAF package was initially added in PR https://github.com/Automattic/jetpack/pull/38332 as part of the PT: Add value to the Protect product card in My Jetpack: pbNhbs-aP6-p2

## Does this pull request change what data or activity we track or use?


No

## Testing instructions:






* On a Jurassic Ninja site, checkout this branch using the Jetpack Beta Tester plugin on the Boost stand-alone plugin (Don't include Jetpack, we want only a stand-alone plugin (that is not Jetpack Protect) to be active).
* Go to the My Jetpack page.
* Open the browser developer console.
* View the value of `window.myJetpackInitialState`. (in the console, type `window.myJetpackInitialState` and press enter).
* Verify that the protect-->wafConfig object (in myJetpackInitialState) only shows a `blocked_logins:` property and does not include other properties (it does not include the WAF config data); See screenshot:
<img width="1760" alt="Screen Shot 2024-08-17 at 10 09 39" src="https://github.com/user-attachments/assets/32bcf6fb-78e9-4959-8feb-727d06793f09">
* Now open the Beta tester plugin and checkout this branch on the Jetpack plugin.
* Open the My Jetpack page again and open the developer console.
* Again view the value of `window.myJetpackInitialState`.
* Verify that the protect-->wafConfig object now shows many other WAF related properties and values; See screenshot:
<img width="1756" alt="Screen Shot 2024-08-17 at 10 14 11" src="https://github.com/user-attachments/assets/d56b97bb-484d-4c04-9b37-ca44c6d12673">
    * (This is expected because the Jetpack plugin does include the WAF package in it's dependencies (which is fine)).
* Now remove Jetpack (Open Plugins page, deactivate/remove Jetpack), and checkout this branch on the Protect plugin (using the Beta Tester plugin).
* Open the My Jetpack page again and open the developer console.
* Again view the value of `window.myJetpackInitialState`.
* Verify that the protect-->wafConfig object still shows all the WAF related properties and values; See screenshot:
<img width="1756" alt="Screen Shot 2024-08-17 at 10 14 11" src="https://github.com/user-attachments/assets/d56b97bb-484d-4c04-9b37-ca44c6d12673">
* Connect Jetpack. (Site connection is enough, but you can user connect too, if you want)
* Open the Protect page. Click the "Firewall" tab to open the firewall settings.
* Click the "Upgrade to enable automatic firewall protection" link.
* Using credits, go ahead and purchase the Jetpack Scan Daily product. Once purchased, you should be redirected back to the Protect firewall page.
* Toggle On the Automatic Firewall Protection.
* Open the My Jetpack page.
* View the Protect product card, the Auto-firewall column should be showing "On". See Screenshot:
<img width="748" alt="Screen Shot 2024-08-17 at 10 53 52" src="https://github.com/user-attachments/assets/76261220-199a-48c3-bf48-42f0d4e93cf8">
* Open the developer console and view the value of `window.myJetpackInitialState`.
* Verify that the protect-->wafConfig object still shows all the WAF related properties and values and that `automatic_rules_available` is `true` and `jetpack_waf_automatic_rules` is `true`.  : See screenshot:
<img width="1156" alt="Screen Shot 2024-08-17 at 10 59 21" src="https://github.com/user-attachments/assets/79a2c97d-59fc-4f00-b3ad-5cb3dc752f01">
